app/ovpn/servers_recommendations.sh

```diff
@@ -23,17 +20,51 @@ if [[ ! -v SERVER ]]; then
             if [ -f "$JSON_FILE_SERVER_COUNTRIES" ]
                 then
                     echo "$(adddate) INFO: The country codes are known, skipping"
-                    export COUNTRY_CODE=$(cat $JSON_FILE_SERVER_COUNTRIES | jq '.[]  | select(.code == "'${COUNTRY^^}'") | .id')
-                else 
+                else
                     echo "$(adddate) INFO: The country codes are unknown, getting country codes from API"
                     curl -s https://nordvpn.com/wp-admin/admin-ajax.php?action=servers_countries -o /tmp/servers_countries
-                    export COUNTRY_CODE=$(cat $JSON_FILE_SERVER_COUNTRIES | jq '.[]  | select(.code == "'${COUNTRY^^}'") | .id')
             fi
-          
-          #GET fastest server based on COUNTRY
-          #https://nordvpn.com/wp-admin/admin-ajax.php?action=servers_recommendations&filters={%22country_id%22:106}
-          wget --quiet --header 'cache-control: no-cache' --output-document=$JSON_FILE ''$SERVER_RECOMMENDATIONS_URL'&filters={%22country_id%22:'$COUNTRY_CODE'}'
+
+            export COUNTRY_CODE=$(cat $JSON_FILE_SERVER_COUNTRIES | jq '.[]  | select(.code | ascii_upcase == "'${COUNTRY^^}'") | .id')
+            PARAMS=$PARAMS'&filters={%22country_id%22:'$COUNTRY_CODE'}'
```

Comment:
yea, looking at this structure now.  I'm sure the way the filters are being appended is the issue.  Will get commit pushed to resolve this tonight / tomorrow.

Reply:
Tried really hard to do 

```
 FILTERS=$( jq -c -n \
     --argjson server_type $(($SERVER_TYPE_CODE)) \
     --argjson country_code $(($COUNTRY_CODE)) \
     '{ server_groups: [$server_type], country_id: $country_code }')
```

and inject that into the url for curl.  but couldn't get the formatting right.  So, to get it working, I'm just manually building the json payload.  Wish they had done normal query params, but not my API.


Perhaps you can figure out a way to utilize the above.  It'd make any future extensions a lot easier.